docs/cip/README.md

```diff
@@ -4,704 +4,2800 @@ Title: Ouroboros Leios - Greater transaction throughput
+[adaptive EB production](#adaptive-eb-production) criteria. The RB is a standard
+Praos block with extended header fields to reference one EB and announce another
+EB when created. The optional EB is a larger block containing references to
+additional transactions. The RB chain continues to be distributed exactly as in
+Praos, while Leios introduces a separate mechanism to distribute the same
+headers for rapid EB discovery and <a href="#equivocation">equivocation
+detection</a>.
+
+<a id="rb-header-diffusion" href="#rb-header-diffusion"></a>**RB Header
+Diffusion**: RB headers diffuse independently of standard ChainSync (steps 2a
+and 2b). This separate mechanism enables rapid EB discovery within the strict
+timing bound $\Delta_\text{hdr}$. Headers are diffused freshest-first to
+facilitate timely EB delivery, with nodes propagating at most two headers per
+(slot, issuer) pair to detect <a href="#equivocation">equivocation</a> - where
+an attacker creates multiple EBs for the same block generation opportunity -
+while limiting network overhead. The header contains the EB hash when the block
+producer created an EB, allowing peers to discover the corresponding EB.
+
+<a id="rb-body-diffusion" href="#rb-body-diffusion"></a>**RB Body Diffusion**:
+After receiving headers, nodes fetch RB bodies via standard BlockFetch protocol
+(step 3). This employs ChainSync and BlockFetch protocols without modification
+for fetching complete ranking blocks after headers are received. The pipelining
+and batching optimizations for block body transfer remain unchanged from Praos.
+
+<a id="rb-validation-adoption" href="#rb-validation-adoption"></a>**Validation
+and Adoption**: Nodes validate the RB and any included EB certificate before
+adopting the block (step 4). This includes cryptographic verification of
+certificates and ensuring they correspond to properly announced EBs. The
+complete validation procedure is detailed in
+[certificate validation](#certificate-validation). The node serves RBs to
+downstream peers using standard Praos block distribution mechanisms (step 5),
+which are permitted to include diffusion pipelining with delayed validation.
+
+#### EB Diffusion
+
+Whenever an EB is announced through an RB header, nodes must fetch the EB
+content promptly (step 6), such that they receive it within
+$L_\text{equi} + L_\text{vote}$ and consequently enables them to vote. EBs are
+fetched freshest-first to ensure timely delivery within the voting window. Only
+the EB body corresponding to the first EB announcement/RB header received for a
+given RB creation opportunity shall be downloaded. The EB contains references to
+transactions.
+
+<a id="eb-chain-selection" href="#eb-chain-selection"></a>**EB Propagation for
+Chain Selection**: To support efficient chain selection, nodes must receive
+**all EBs from competing forks**, not only those in their current preferred
+chain. This ensures that when a node switches to a different fork due to the
```

Comment:
Should we mention that a node does not have to fetch EBs from forks that have diverged from the locallypreferred chain older than the Praos security parameter?